src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/OSPlatform.cs

```diff
@@ -40,22 +46,19 @@ public static OSPlatform Create(string osPlatform)
 
         public bool Equals(OSPlatform other)
         {
-            return Equals(other._osPlatform);
-        }
-
-        internal bool Equals(string? other)
-        {
-            return string.Equals(_osPlatform, other, StringComparison.Ordinal);
+            return string.Equals(_osPlatform, other._osPlatform, StringComparison.OrdinalIgnoreCase);
         }
 
         public override bool Equals(object? obj)
         {
-            return obj is OSPlatform && Equals((OSPlatform)obj);
+            Debug.Fail("The non generic method should not be used by BCL");
```

Comment:
We want the debug libraries to be fully usable. This should be removed.

If you would like to check that this method is not called by the BCL, it is a job for the analyzer.